geoportailv3/static/js/layerinfo/layerinfoservice.js

```diff
@@ -14,29 +15,91 @@ app.ShowLayerinfo;
 
 
 /**
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
  * @param {angular.$http} $http Angular $http service
  * @param {angular.$sce} $sce Angular $sce service
  * @param {ngeo.CreatePopup} ngeoCreatePopup Ngeo popup factory service
  * @ngInject
  */
-app.showLayerinfoFactory = function($http, $sce, ngeoCreatePopup) {
+app.showLayerinfoFactory = function(gettextCatalog,
+                                    $http,
+                                    $sce,
+                                    ngeoCreatePopup) {
 
   /**
    * @type {ngeo.Popup}
    */
   var popup = ngeoCreatePopup();
+  /**
+   * @type {Object.<!string, Object.<string, !angular.$q.Promise>>}
```

Comment:
Instead of using an object whose values are objects, you could use something like `metadataUid + '##' + currentLanguage` as the key.